pkg/reloader/reloader.go

```diff
@@ -125,8 +128,19 @@ func (r *Reloader) apply(ctx context.Context) error {
 			return errors.Wrap(err, "hash file")
 		}
 		cfgHash = h.Sum(nil)
+		if r.gunzipDir != "" {
+			if err := extract(r.cfgFile, r.gunzipDir); err != nil {
```

Comment:
I think it's wrong as we are watching this file so extracting only on beginning is not enough. I proposed alternative flow in my comment above.